pkgs/shells/oil/default.nix

```diff
@@ -1,42 +1,163 @@
+  # rev == present HEAD of release/0.9.3
+  rev = "20d23eed7949548bc897ddcf7701f720c81f438e";
+  src = fetchFromGitHub {
+    owner = "oilshell";
+    repo = "oil";
+    inherit rev;
+    hash = "sha256-QNSISr719ycZ1Z0quxHWzCb3IvHGj9TpogaYz20hDM4=";
+
+    /*
+      It's not critical to drop most of these; the primary target is
+      the vendored fork of Python-2.7.13, which is ~ 55M and over 3200
+      files, dozens of which get interpreter script patches in fixup.
 
-  src = fetchurl {
-    url = "https://www.oilshell.org/download/oil-${version}.tar.xz";
-    sha256 = "sha256-YvNgcvafM3jgO3nY1SVcHRNglOwRQQ208W7oLxZg79o=";
+      Note: -f is necessary to keep it from being a pain to update
+      hash on rev updates. Command will fail w/o and not print hash.
+    */
+    # extraPostFetch = ''
+    #   rm -rf benchmarks metrics py-yajl rfc gold web testdata services demo devtools cpp
+    # '';
+  };
+  re2c = stdenv.mkDerivation rec {
+    pname = "re2c";
+    version = "1.0.3";
+    sourceRoot = "${src.name}/re2c";
+    src = fetchFromGitHub {
+      owner = "skvadrik";
+      repo = "re2c";
+      rev = version;
+      sha256 = "0grx7nl9fwcn880v5ssjljhcb9c5p2a6xpwil7zxpmv0rwnr3yqi";
+    };
+    nativeBuildInputs = [ autoreconfHook ];
+    preCheck = ''
+      patchShebangs run_tests.sh
+    '';
   };
+  py-yajl = python27Packages.buildPythonPackage rec {
+    pname = "oil-pyyajl-unstable";
+    version = "2019-12-05";
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "py-yajl";
+      rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+      sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+      fetchSubmodules = true;
+    };
+    # just for submodule IIRC
+    nativeBuildInputs = [ git ];
+  };
+  patchSrc = fetchFromGitHub {
+    owner = "abathur";
+    repo = "nix-py-dev-oil";
+    rev = "v0.8.12.1";
+    hash = "sha256-7JVnosdcvmVFN3h6SIeeqcJFcyFkai//fFuzi7ThNMY=";
+  };
+in
 
-  postPatch = ''
-    patchShebangs build
-  '';
+  /*
+    Upstream isn't interested in packaging this as a library
+    (or accepting all of the patches we need to do so).
+    This creates one without disturbing upstream too much.
+  */
+stdenv.mkDerivation {
+  pname = "oil";
+  inherit version src;
+
+  # patch to support a python package, pass tests on macOS, etc.
+  patches = [
+    "${patchSrc}/0004-disable-internal-py-yajl-for-nix-built.patch"
+    "${patchSrc}/0006-disable_failing_libc_tests.patch"
+  ];
+
+  # TODO substitue the naked python2 call instead of having python27 as a dependency
+  # TODO fix pythonpath to include py-yajl
+  buildInputs = [ readline cmark py-yajl cmake python27Full ];
 
-  preInstall = ''
-    mkdir -p $out/bin
+  nativeBuildInputs = [ re2c file makeWrapper ];
+
+  configurePhase = ''
+    build/dev.sh yajl-release
+    build/prepare.sh configure
+    build/prepare.sh build-python
+    build/dev.sh all
+  '';
+  buildPhase = ''
+    make _bin/oil.ovm-dbg
+    make _release/oil.tar
```

Comment:
Those probably should have -J$NIX_BUILD_CORES to respect the nix setting.

Reply:
Thank you!